src/main/java/org/rumbledb/items/parsing/ItemParser.java

```diff
@@ -274,6 +275,58 @@ private static void addValue(
         }
     }
 
+    public static DataType getDataFrameDataTypeFromItemTypeName(String itemTypeName) {
+        switch (itemTypeName) {
+            case "boolean":
+                return DataTypes.BooleanType;
+            case "integer":
+                return DataTypes.IntegerType;
+            case "double":
+                return DataTypes.DoubleType;
+            case "decimal":
+                return DataTypes.LongType;
+            case "string":
+                return DataTypes.StringType;
+            case "null":
+                return DataTypes.NullType;
+            case "date":
+                return DataTypes.DateType;
+            case "datetime":
+                return DataTypes.TimestampType;
+            case "hexbinary":
+                return DataTypes.BinaryType;
+            case "[\"item\"]":
+                return vectorType;
+            default:
+                throw new IllegalArgumentException("Unexpected item type found: '" + itemTypeName + "'.");
+        }
+    }
+
+    public static String getItemTypeNameFromDataFrameDataType(DataType dataType) {
+        if (DataTypes.BooleanType.equals(dataType)) {
+            return "boolean";
+        } else if (DataTypes.IntegerType.equals(dataType) || DataTypes.ShortType.equals(dataType)) {
+            return "integer";
+        } else if (DataTypes.DoubleType.equals(dataType) || DataTypes.FloatType.equals(dataType)) {
+            return "double";
+        } else if (DataTypes.LongType.equals(dataType)) {
+            return "decimal";
+        } else if (DataTypes.StringType.equals(dataType)) {
+            return "string";
+        } else if (DataTypes.NullType.equals(dataType)) {
+            return "null";
+        } else if (DataTypes.DateType.equals(dataType)) {
+            return "date";
+        } else if (DataTypes.TimestampType.equals(dataType)) {
+            return "datetime";
+        } else if (DataTypes.BinaryType.equals(dataType)) {
+            return "hexbinary";
+        } else if (vectorType.equals(dataType)) {
+            return "[\"item\"]";
```

Comment:
If my understanding is correct that this is also only used to check the dataframe schema types against the types passed to annotate, this is fine too (also not documented).